src/utils/__tests__/changelog.test.ts

```diff
@@ -304,391 +281,346 @@ describe('generateChangesetFromGit', () => {
-    `);
-  });
-
-  it('should escape # signs on milestone titles', async () => {
-    mockGetChangesSince.mockResolvedValueOnce([
-      {
-        hash: 'abcdef1234567890',
-        title: 'Upgraded the kernel',
-        body: '',
-        pr: '123',
-      },
-    ]);
-    makeCommitResponse([
+      [
+        '### Better drivetrain',
+        '',
+        'We have upgraded the drivetrain for a smoother and more performant driving experience. Enjoy!',
+        '',
+        'PRs: #123, #456',
+        '',
+        '### Better driver experience (ongoing)',
+        '',
+        'PRs: #789, #900',
+        '',
+        '### Various fixes & improvements',
+        '',
+        '- Upgraded the kernel (abcdef12)',
+        '- Fix the clacking sound on gear changes (#950)',
+      ].join('\n'),
+    ],
+    [
+      'should escape # signs on milestone titles',
+      [
+        {
+          hash: 'abcdef1234567890',
+          title: 'Upgraded the kernel',
+          body: '',
+          pr: {
+            local: '123',
+            remote: {
+              number: '123',
+              milestone: '1',
+            },
+          },
+        },
+      ],
       {
-        hash: 'abcdef1234567890',
-        pr: '123',
-        milestone: '1',
-      },
-    ]);
-    mockClient.mockResolvedValueOnce({
-      repository: {
-        M1: {
+        '1': {
           title: 'Drivetrain #1 in town',
           description:
             'We have upgraded the drivetrain for a smoother and more performant driving experience. Enjoy!',
           state: 'CLOSED',
         },
       },
-    });
-    const changes = await generateChangesetFromGit(dummyGit, '1.0.0');
-    expect(changes).toMatchInlineSnapshot(`
-      "### Drivetrain &#35;1 in town
-
-      We have upgraded the drivetrain for a smoother and more performant driving experience. Enjoy!
-
-      PRs: #123"
-    `);
-  });
-  it('should omit milestone body if it is empty', async () => {
-    mockGetChangesSince.mockResolvedValueOnce([
-      {
-        hash: 'abcdef1234567890',
-        title: 'Upgraded the kernel',
-        body: '',
-        pr: '123',
-      },
-    ]);
-    makeCommitResponse([
+      [
+        '### Drivetrain &#35;1 in town',
+        '',
+        'We have upgraded the drivetrain for a smoother and more performant driving experience. Enjoy!',
+        '',
+        'PRs: #123',
+      ].join('\n'),
+    ],
+    [
+      'should omit milestone body if it is empty',
+      [
+        {
+          hash: 'abcdef1234567890',
+          title: 'Upgraded the kernel',
+          body: '',
+          pr: { local: '123', remote: { number: '123', milestone: '1' } },
+        },
+      ],
       {
-        hash: 'abcdef1234567890',
-        pr: '123',
-        milestone: '1',
-      },
-    ]);
-    mockClient.mockResolvedValueOnce({
-      repository: {
-        M1: {
+        '1': {
           title: 'Better drivetrain',
           state: 'CLOSED',
         },
       },
```

Comment:
What `body` does GitHub return on an empty milestone body? We're assuming the body to be trimmed, but does this mean it's not defined, or to be `''`? In a [past PR](https://github.com/getsentry/craft/pull/296#discussion_r715568597) this was discussed when `description` was `''`. Bodies, when empty, they are empty strings, why? Or, why milestones aren't?

Reply:
You just get a `null` in place of the milestone. The `description` comes as an empty string when it is not filled. I'll make the adjustment in code to clarify it.